### ISSUE TITLE
Could you explain this code?

### ISSUE THREAD

**Issue by awrm20** [praeclarumjj3/VQ-VAE-on-MNIST#3]
<img width="296" alt="Screenshot 2021-07-01 203046" src="https://user-images.githubusercontent.com/86297999/124110245-61341c80-daab-11eb-9bf1-45d085823f34.png">
Sorry for beginner level questions. could you please answer them:
(loss_recons) is reconstruction error or reconstruction loss?
How we can find accuracy of the model and validate our results?
In last line we are taking average of the loss any significance?

**Comment by praeclarumjj3** [comment]
loss_recon is the Reconstruction Loss that is **squared the difference of the predicted image and real image**. You can also call it Reconstruction Error, doesn't make a difference.

This is not a classification task. It's a Generative Modeling Problem, so there's no metric like accuracy here as there are no labels. However if you do want accuracy, you can pass the generated images through a classification model and compare the labels then to get a sense about the realistic nature of images. That's what happens when we train a GAN. Still, it doesn't make sense to talk about **accuracy** in case of image generation tasks.

We are iterating through batches of images (say N images in a batch) and adding the loss for each batch. So at the end we have loss corresponding to the number of batches. However we want loss per image, so we divide the total loss with *len(dataloader)* which is inherently the number of batches.
